### PR TITLE
Add positive Lasso solvers 

### LINKED ISSUE #50: ENH: add support for positive lasso?
I believe it is OK, we just need to modify the following lines:

https://github.com/mathurinm/celer/blob/master/celer/dense.pyx#L39

https://github.com/mathurinm/celer/blob/master/celer/dense.pyx#L58

https://github.com/mathurinm/celer/blob/master/celer/dense.pyx#L424

+ add a `positive` parameter to the 3 cython functions and to the wrapper and Lasso/LassoCV

testing it should also straightforward: we just have to compare with sklearn

@hichamjanati

### PR DESCRIPTION
**Pull request by hichamjanati**
Closes #50

### CODE REVIEW

File: celer/dense.pyx [hunk 1]
```diff
@@ -31,12 +32,16 @@ cdef floating compute_dual_scaling_dense(int n_samples, int n_features,
...
+            Xj_theta = fdot(&n_samples, &theta[0], &inc, &X[0, j], &inc)
+            if positive == 0.:
+                Xj_theta = fabs(Xj_theta)
             scal = max(scal, Xj_theta)
     else: # scaling wrt features in C only
         for j in range(ws_size):
             Cj = C[j]
-            Xj_theta = fabs(fdot(&n_samples, &theta[0], &inc, &X[0, Cj], &inc))
+            Xj_theta = fdot(&n_samples, &theta[0], &inc, &X[0, Cj], &inc)
+            if positive == 0.:
```
Comment (mathurinm) [inline celer/dense.pyx]: I believe `if not positive` is the pythonic way. WDYT @hichamjanati

File: celer/dense.pyx [hunk 2]
```diff
@@ -31,12 +32,16 @@ cdef floating compute_dual_scaling_dense(int n_samples, int n_features,
     cdef int inc = 1
     if ws_size == n_features: # scaling wrt all features
         for j in range(n_features):
-            Xj_theta = fabs(fdot(&n_samples, &theta[0], &inc, &X[0, j], &inc))
+            Xj_theta = fdot(&n_samples, &theta[0], &inc, &X[0, j], &inc)
+            if positive == 0.:
```
Comment (mathurinm) [inline celer/dense.pyx]: ditto

File: celer/dense.pyx (hunk 1 above)
Comment (hichamjanati) [reply celer/dense.pyx]: Since positive is boolean, I agree @mathurinm

File: .pytest_cache/v/cache/lastfailed [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (mathurinm) [inline .pytest_cache/v/cache/lastfailed]: you can add .pytest_cache/* to .gitignore and remove this file @hichamjanati

File: .pytest_cache/v/cache/nodeids [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[
```
Comment (mathurinm) [inline .pytest_cache/v/cache/nodeids]: remove also

File: celer/dense.pyx [hunk 5]
```diff
@@ -56,7 +61,10 @@ cdef void set_feature_prios_dense(int n_samples, int n_features, floating[:] the
             prios[j] = 10000
             continue
         Xj_theta = fdot(&n_samples, &theta[0], &inc, &X[0, j], &inc)
-        prios[j] = fabs(fabs(Xj_theta) - 1.) / norms_X_col[j]
+        if positive:
+            prios[j] = fabs(Xj_theta - 1.) / norms_X_col[j]
```
Comment (mathurinm) [inline celer/dense.pyx]: @hichamjanati small question, the constraint j on theta is `0. <= Xj_theta <= 1` (as opposed to -1 <= ...<= 1 for the Lasso) ?

Comment (mathurinm) [reply celer/dense.pyx]: or rather  `Xj_theta <= 1` ?

Comment (hichamjanati) [reply celer/dense.pyx]: @mathurinm  yes Xj_theta <= 1

**Comment by hichamjanati** [comment]
We can merge, WDYT @mathurinm ?

**Comment by mathurinm** [comment]
Thank you @hichamjanati !

**Pull request merged**